src/register-subscriptions.ts

```diff
@@ -0,0 +1,149 @@
+    return beacon.chains.map(async (chain) => {
+      const chainId = chainNameToChainId[chain.name];
+      if (!chainId) throw new Error(`🛑 Unknown chain name: ${chain.name}`);
+
+      if (!chainNameToPublicRPCUrl[chain.name]) throw new Error(`🛑 No public RPC URL for chain ${chain.name}`);
+      const provider = new ethers.providers.JsonRpcProvider(chainNameToPublicRPCUrl[chain.name]);
+      const DapiServerAddress = operationsRepository.documentation.chains[chain.name].contracts['DapiServer'];
+      if (!DapiServerAddress) throw new Error(`🛑 No DapiServer contract address for chain ${chain.name}`);
+      const DapiServer = DapiServerContract(DapiServerAddress, provider);
+
+      const sponsor = chain.sponsor;
+
+      const expectedSubscriptionId = ethers.utils.keccak256(
+        ethers.utils.defaultAbiCoder.encode(
+          ['uint256', 'address', 'bytes32', 'bytes', 'bytes', 'address', 'address', 'address', 'bytes4'],
+          [
+            chainId,
+            airnodeAddress,
+            templateId,
+            parameters,
+            encodedBeaconUpdateSubscriptionConditions,
+            airnodeAddress,
+            sponsor,
+            DapiServerAddress,
+            DapiServerInteface.getSighash('fulfillPspBeaconUpdate'),
+          ]
+        )
+      );
+
+      // eslint-disable-next-line no-async-promise-executor
+      return new Promise(async (resolve, reject) => {
+        try {
+          console.log(`🔗 Registering subscriptionId for beacon ${beaconName} on chain ${chain.name}`);
+
+          const registerBeaconUpdateSubscription = await DapiServer.connect(
+            nonceManagers[chain.name]
+          ).registerBeaconUpdateSubscription(
+            airnodeAddress,
+            templateId,
+            encodedBeaconUpdateSubscriptionConditions,
+            airnodeAddress,
+            sponsor
+          );
+
+          // Check that the transaction is complete
+          const tx = await registerBeaconUpdateSubscription.wait();
+
+          const subscriptionId = tx.events.find(
+            (event: { event: string }) => event.event === 'RegisteredBeaconUpdateSubscription'
+          ).args.subscriptionId;
+
+          if (subscriptionId !== expectedSubscriptionId) {
+            reject(`🛑 The subscription ID ${subscriptionId} does not match the expected ID ${expectedSubscriptionId}`);
+          }
+
+          resolve(
+            `✅ Subscription registered with ID ${subscriptionId} for beacon ${beaconName} on chain ${chain.name}`
+          );
+        } catch (error) {
+          console.error(`🛑 Error registering subscription for beacon ${beaconName} on chain ${chain.name}`);
+          reject(error);
+        }
+      });
+    });
+  });
+
+  await Promise.allSettled(subscriptionPromises).then((results) => {
+    results.forEach((result) => {
+      if (result.status === 'fulfilled') {
+        console.log(result.value);
+      } else {
+        console.log(result.reason);
+      }
+    });
+  });
```

Comment:
This also doesn't really make sense, it should rather be handled like this:
```suggestion
  const results = await Promise.allSettled(subscriptionPromises);

  results.forEach((result) => {
    if (result.status === 'fulfilled') {
      console.log(result.value);
    } else {
      console.log(result.reason);
    }
  });
```
It could be handled `sync`, but then the `await` should be removed.

Reply:
my bad fixed it